pynest/examples/pynest-microcircuit/network_params.py

```diff
@@ -0,0 +1,192 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+'''
+pynest microcicuit parameters
+----------------------------------------------
+
+Network parameters for the microcircuit.
+
+Hendrik Rothe, Hannah Bos, Sacha van Albada; May 2016
+'''
+
+import numpy as np
+
+
+def get_mean_delays(mean_delay_exc, mean_delay_inh, number_of_pop):
+    """Creates matrix containing the delay of all connections.
+
+    Arguments:
+    mean_delay_exc: delay of the excitatory connections
+    mean_delay_inh: delay of the inhibitory connections
+    number_of_pop: number of populations
+
+    Returns:
+    mean_delays: matrix specifying the mean delay of all connections
+    """
+
+    dim = number_of_pop
+    mean_delays = np.zeros((dim, dim))
+    mean_delays[:, 0:dim:2] = mean_delay_exc
+    mean_delays[:, 1:dim:2] = mean_delay_inh
+    return mean_delays
+
+
+def get_std_delays(std_delay_exc, std_delay_inh, number_of_pop):
+    """Creates matrix containing the standard deviations of all delays.
+
+    Arguments
+    std_delay_exc: standard deviation of excitatory delays
+    std_delay_inh: standard deviation of inhibitory delays
+    number_of_pop: number of populations in the microcircuit
+
+    Returns:
+    std_delays: matrix specifying the standard deviation of all delays
+    """
+
+    dim = number_of_pop
+    std_delays = np.zeros((dim, dim))
+    std_delays[:, 0:dim:2] = std_delay_exc
+    std_delays[:, 1:dim:2] = std_delay_inh
+    return std_delays
+
+
+def get_mean_PSP_matrix(PSP_e, g, number_of_pop):
+    """This function creates a matrix of the mean evoked postsynaptic
+    potential between the recurrent connections.
+
+    Arguements:
+    PSP_e: mean evoked potential in mV
+    g: relative strength of the inhibitory to excitatory connection
+    number_of_pop: number of populations in the microcircuit
+
+    Returns:
+    Matrix of the postsynaptic potentials
+    """
+    dim = number_of_pop
+    weights = np.zeros((dim, dim))
+    exc = PSP_e
+    inh = PSP_e * g
+    weights[:, 0:dim:2] = exc
+    weights[:, 1:dim:2] = inh
+    # The weight of the connection L23E to L4E is doubled.
+    weights[0, 2] = exc * 2
+    return weights
+
+
+def get_std_PSP_matrix(PSP_rel, number_of_pop):
+    """This function creates a matrix of the standard deviation of the
+     evoked postsynaptic potential.
+
+    Arguements:
+    PSP_rel: standard deviation of the evoked synaptiv potential
+    number_of_pop: number of populations in the microcircuit
+
+    Returns:
+    Matrix of the postsynaptic potentials.
+    """
+    dim = number_of_pop
+    std_mat = np.zeros((dim, dim))
+    std_value_normal = PSP_rel
+    std_value_exception = 0.05
+    std_mat[:, :] = std_value_normal
+    std_mat[0, 2] = std_value_exception
```

Comment:
What is the reason for doing this? We find this neither in the SLI implementation nor in the paper.